plugins/catalog-backend/src/catalog/DatabaseCatalog.test.ts

```diff
@@ -38,15 +38,23 @@ describe('DatabaseCatalog', () => {
     });
   });
 
-  it('instantiates', () => {
+  it('manages locations', async () => {
     const catalog = new DatabaseCatalog(database, logger);
-    expect(catalog).toBeDefined();
-  });
+    const input = { type: 'a', target: 'b' };
+    const output = { id: expect.anything(), type: 'a', target: 'b' };
 
-  describe(`refreshLocations`, () => {
-    it('works with no locations added', async () => {
-      const catalog = new DatabaseCatalog(database, logger);
-      await expect(catalog.refreshLocations()).resolves.toBeUndefined();
-    });
+    await catalog.addLocation(input);
+
+    const locations = await catalog.locations();
+    expect(locations).toEqual([output]);
+    const location = await catalog.location(locations[0].id);
+    expect(location).toEqual(output);
+
+    await catalog.removeLocation(locations[0].id);
+
+    await expect(catalog.locations()).resolves.toEqual([]);
```

Comment:
How about `expect(await catalog.locations()).toEqual([])` ?

Reply:
Yepyep, that's functionally equal, but then we have to use the other pattern anyway for `.rejects` so we don't have to wrap it in a try/catch just to use await! So I went for making it this way, plus i kind of like how it reads in terms of plain english - "expect that this resolves to equal the empty list" - very purpose driven :)

Reply:
Yeah you'll get a better error message if this fails, so +1 to using this